src/restore.ts

```diff
@@ -35,22 +35,46 @@ async function run(): Promise<void> {
             restoreKeys
         );
 
+        //Check if user wants to save cache despite of failure in any previous job
+        const saveCache = core.getInput(Inputs.SaveOnAnyFailure).toLowerCase();
```

Comment:
Now that we have two different actions, user can ensure this by calling `save` action explicitly and using a condition of `if: always()`. I think this is same as what David had mentioned in ADR comment.

Reply:
Essentially what I am saying is that the original `cache` action behavior remains same. User should use new `restore` and `save` action with right conditionals if they want to have better control like save on failure, or fail on cache miss etc

Reply:
Then do we still need this input?

Reply:
Yes, `if: always()` should be used with `save` action.
and `Inputs.SaveOnAnyFailure` input should be used in the `cache` action input.

Reply:
Okay, so intention was not clear from the code. Another reason why we need to have separate wrappers for `restore`, `save` and `cache` actions otherwise it going to become unmanageable.

Reply:
Also, what if we do not provide this "save on failure" functionality in cache action and ask users to use the two new actions if they want this. IMO we should not change the behaviour of cache action much for such restore-save dependency related scenarios, and let the two new actions take care of such.
Thoughts?

Reply:
Oh okay, in short we only want to use the `if: always()` with `save` action and not change anything in the `cache` action.
But let's say there's some failure even before anything related to build or cache restore happens, for example the `checkout` action fails, in that case too we will execute the save action, which won't be a desired behaviour right?

Reply:
Good point. Can user use `if` conditional to run only if restore was success?
Something like `if: always() && steps.<restore-step>. outcome  == 'success'`? 
Not sure if `steps` context is available in `if`. Worth trying.

Reply:
Looks like it may be available, so definitely worth trying.

Reply:
Btw please note that I am not totally against these additional inputs but they do not feel like must-to-do. Hence better to keep things simple for now and add later if there are additional asks.